internal/executor/instance/instance.go

```diff
@@ -73,6 +74,15 @@ func NewFromProto(anyInstance *any.Any, commands []*api.Command) (Instance, erro
 			Arguments:  instance.Arguments,
 		}, nil
 	case *api.PersistentWorkerInstance:
+		return NewPersistentWorkerInstance()
+	case *api.DockerBuilder:
+		// Ensure that we're not trying to run e.g. Windows-specific scripts on macOS
+		instanceOS := strings.ToLower(instance.Platform.String())
+		if runtime.GOOS != instanceOS {
```

Comment:
you can run linux docker builder on macOS, don't you?

Reply:
I don't think this is a great idea, because this might result in a very surprising behavior. But should it?

Besides, the Docker Builder itself might be somewhat surprising: imagine scripts like `rm -rf /tmp` (or more important paths) will run on the host system.